lib/src/data/mixins/error_to_failure_mixin.dart

```diff
@@ -0,0 +1,64 @@
+import 'package:dio/dio.dart';
+import 'package:either_dart/either.dart';
+import 'package:q_architecture/q_architecture.dart';
+
+mixin ErrorToFailureMixin {
+  EitherFailureOr<T> execute<T>(
+    EitherFailureOr<T> Function() function, {
+    required ErrorResolver errorResolver,
+  }) async {
+    try {
+      return await function();
+    } catch (err, stackTrace) {
+      return Left(errorResolver.resolve(err, stackTrace));
+    }
+  }
+}
+
+abstract interface class ErrorResolver {
+  Failure resolve<T>(Object error, [StackTrace? stackTrace]);
+}
+
+class ApiErrorResolver implements ErrorResolver {
+  final Map<int, Failure> failures;
+
+  const ApiErrorResolver({
+    required this.failures,
+  });
+
+  @override
+  Failure resolve<T>(Object error, [StackTrace? stackTrace]) {
+    if (error is! DioException)
+      return Failure.generic(
+        error: error,
+        stackTrace: stackTrace,
+      );
+    final response = error.response;
+    final key =
+        failures.keys.firstWhereOrNull((code) => code == response?.statusCode);
+    return failures[key] ??
+        Failure.generic(
+          error: error,
+          stackTrace: stackTrace,
+        );
+  }
+
+  @override
+  // ignore: override_on_non_overriding_member
```

Comment:
Am I missing something here or this ApiErrorResolver needs to extend Equatable to be able to override props and stringify?